### DISCUSSION TITLE
Understanding the reasoning behind "WhereUniqueInput"?

### DISCUSSION THREAD

**Discussion by florianmartens** [prisma/prisma#10337] [Q&A] [answered]
I find it a little hard to grasp why query operations on individual elements are constrained to unique fields. Let's look at an example:

`prisma.Post.delete({where: {id: 1}})`

For simple cases this seems sufficient. But what happens if we want to ensure that the posts belong to the user issuing the request. We'd need to change the query:

`prisma.Post.delete({where: { id: 1, author: {id: user.id}}}) // -> does not work`

But this is not possible. I'm stuck with writing two queries (1: check if the user is the author, 2: delete the post) or use an operation like `deleteMany`. The downside to (...)Many operations is that they don't return the instance but a list or object with a `count` property.

This problem is also super, super relevant for everyone implementing soft-deletes (which should be a big use case for almost all apps). 

I struggle with why this is necessary. I'm guessing that Prisma is just constructing a SQL WHERE clause to which we can add many constraints. I find this to be limiting when working with Prisma and would like to understand why it's implemented this way?

**Comment by austincrim** [comment, marked as answer]
Hey @florianmartens 👋, thanks for the question.

The goal of the `delete` method is to find and remove a **single, unique row**. This is already accomplished when you pass in a unique identifier, it can't really take any more information. Verifying the `userId` feels more like a validation concern, which isn't currently in scope for Prisma. You can read more discussion of this [here](https://github.com/prisma/prisma/issues/7290), please feel free to add your own thoughts and use case!

You can also use `prisma.$executeRaw` if you wanted to write your own SQL query that fits your exact needs.

Let me know if you have any other questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
